### PR TITLE
`MediaContainer` class is now a list

### LINKED ISSUE #1372: `totalSize` of MediaContainer
### What is your feature request?

api returns a bunch of metadata along with items fetched one of which is `totalSize`

it is useful in showing totalItems on the server without fetching all of them

other attributes are `offset` and `size`, standard pagination stuff

### Are there any workarounds?

`fetchItems` already uses this info
https://github.com/pkkid/python-plexapi/blob/abcab4f78d22e86b380327c6014d3148e0248e6d/plexapi/base.py#L258

### Code Snippets

```python
tracks = music_section.searchTracks(maxresults=100)
print(f'showing {len(tracks)} of {tracks.totalSize} starting at {tracks.offset}')
```

### Additional Context

image
as on plex web

### PR DESCRIPTION
**Pull request by Dr-Blank**
## Description

`fetchItems` returns a MediaContainer, exactly how server does and is also a python list! Hence it would not break any current api.
additionally, it now supports totalSize attr

The only breaking change is that MediaContainer can now only accept `initpath` and `parent` as keword-only arguments in its initialisation, which I don't think anyone would be using.

Fixes #1372 
## Type of change

Please delete options that are not relevant.

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated the docstring for new or existing methods
- [x] I have added tests when applicable

### CODE REVIEW

**Comment by Dr-Blank** [comment]
could you guide me as to where to put the test for this?
also how would the test be implemented?
for example test fetchItems directly
or indirectly, like searchepisodes, and test on the return

tests i can think of:
 1. isinstance list
 2. size, offset, totalSize are int

File: plexapi/base.py [hunk 1]
```diff
@@ -1009,11 +1020,55 @@ class MediaContainer(PlexObject):
...
 
+    def __init__(
+        self,
+        server: "PlexServer",
+        data: Element,
+        *args: PlexObjectT,
+        initpath: Optional[str] = None,
+        parent: Optional[PlexObject] = None,
+    ) -> None:
+        # super calls Generic.__init__ which calls list.__init__ eventually
```
Comment (Dr-Blank) [inline plexapi/base.py:1038]: `mro` suggests otherwise, please review

File: plexapi/base.py [hunk 2]
```diff
@@ -1009,11 +1020,55 @@ class MediaContainer(PlexObject):
...
+        # update size, totalSize, and offset
+        if not isinstance(__iterable, MediaContainer):
+            return
+
+        # prefer the totalSize of the new iterable even if it is smaller
+        self.totalSize = (
+            __iterable.totalSize
+            if __iterable.totalSize is not None
+            else self.totalSize
+        )  # ideally both should be equal
```
Comment (Dr-Blank) [inline plexapi/base.py:1057]: newest value will be the latest from the server, but total size should not change in such short time

File: plexapi/base.py [hunk 3]
```diff
@@ -1009,11 +1020,55 @@ class MediaContainer(PlexObject):
...
+
+        # the size of the new iterable is added to the current size
+        self.size = curr_size + (
```
Comment (Dr-Blank) [inline plexapi/base.py:1060]: totalsize get increased as more and more items are fetched

File: plexapi/base.py [hunk 4]
```diff
@@ -1009,11 +1020,55 @@ class MediaContainer(PlexObject):
...
+            __iterable.size if __iterable.size is not None else len(__iterable)
+        )
+
+        # the offset is the minimum of the two, prefering older values
+        if self.offset is not None and __iterable.offset is not None:
+            self.offset = min(self.offset, __iterable.offset)
+        else:
+            self.offset = (
+                self.offset if self.offset is not None else __iterable.offset
+            )
```
Comment (Dr-Blank) [inline plexapi/base.py:1070]: if we start fetching from 0 and fetch 50 items, and next fetch offset is 50, however when etending, the offset is still from 0, but now the size is 50 + 50 =100

**Comment by Dr-Blank** [comment]
I suppose regression tests are not needed for this PR since the current tests should be enough.

I have added some tests pertaining to new functionality only.

**Comment by JonnyWong16** [comment]
This looks greats. It seems to be mostly working, but is it possible to load the other `MediaContainer` attributes as well?

```py
>> results = plex.library.section("Movies (Test)").all()  # results is now a MediaContainer
>> results.librarySectionID  # this is None
```

```xml
<MediaContainer size="17" allowSync="1" art="/:/resources/movie-fanart.jpg" identifier="com.plexapp.plugins.library" librarySectionID="41" librarySectionTitle="Movies (Test)" librarySectionUUID="e9535986-3fd3-4f88-9793-77a50b04dd01" mediaTagPrefix="/system/bundle/media/flags/" mediaTagVersion="1708667686" thumb="/:/resources/movie.png" title1="Movies (Test)" title2="All Movies (Test)" viewGroup="movie" viewMode="65592">
```

`results` is initialized with empty `data = Element('MediaContainer')` which never gets updated with the attributes from the XML.
`subresults` gets initialized with the actual data from the XML (from `items` inside `findItems()`).

**Comment by Dr-Blank** [comment]
> is it possible to load the other `MediaContainer` attributes as well?

https://github.com/pkkid/python-plexapi/pull/1373/commits/06819fa614ce3d9d84f0f51980094cc43569b22e solves this

**Review by JonnyWong16 — APPROVED** [review]

File: plexapi/base.py [hunk 5]
```diff
@@ -324,7 +331,7 @@ def findItems(self, data, cls=None, initpath=None, rtag=None, **kwargs):
         if rtag:
             data = next(utils.iterXMLBFS(data, rtag), [])
         # loop through all data elements to find matches
-        items = []
+        items = MediaContainer[cls](self._server, data, initpath=initpath) if data.tag == 'MediaContainer' else []
```
Comment (JonnyWong16) [inline plexapi/base.py:334]: `AttributeError: 'list' object has no attribute 'tag'` when `data = []` from the line above.

Comment (Dr-Blank) [reply plexapi/base.py:334]: Maybe we can initialise an empty media container when the tag attribute does not exist? I'll create a pr soon if not already done.

Comment (JonnyWong16) [reply plexapi/base.py:334]: Initialize an `empty` element like what you did here:

https://github.com/pkkid/python-plexapi/blob/d6754c0cdeec67056df10318e708ca6ec4a61bb0/plexapi/base.py#L459

Comment (Dr-Blank) [reply plexapi/base.py:334]: should be resolved in #1396

**Pull request merged**